haystack/components/converters/pypdf.py

```diff
@@ -31,7 +31,7 @@ class DefaultConverter:
 
     def convert(self, reader: "PdfReader") -> Document:
         """Extract text from the PDF and return a Document object with the text content."""
-        text = "".join(page.extract_text() for page in reader.pages if page.extract_text())
+        text = "\f".join(page.extract_text() for page in reader.pages)
```

Comment:
I believe it is an empty string when there is no text present. The new unit test covers this case, since the correct number of page breaks is only correct if this is removed.

Also looking at their code their types at least indicate that `page.extract_text()` always has to return a string.

Reply:
Granted their code is hard to follow so I'm not sure if it's impossible for None to occur so we could add handling of that, but I'm not sure if that would be easy to test.

Reply:
I did a quick check and agree with you that we should always get a string.
https://github.com/py-pdf/pypdf/blob/fc893d5d751db77a65370555f32232be4b68e451/pypdf/_page.py#L1546